test_scripts/Defects/8_1/3838_3_SDL_does_not_send_UI.ClosePopUp_in_case_two_PerformInteraction_requests.lua

```diff
@@ -0,0 +1,140 @@
+-- 2. Mobile app requests PerformInteraction(VR_ONLY) one more time while the first one is in progress
+-- 3. HMI responds with 'REJECTED' to second UI/VR.PerformInteraction requests
+-- SDL does:
+-- - not send UI.ClosePopUp request to HMI
+-- - send PerformInteraction response with (resultCode: REJECTED, success:false) to mobile App
+-- 4. HMI responds with 'SUCCESS' to first UI/VR.PerformInteraction requests
+-- SDL does:
+-- - send PerformInteraction response with choiceID from VR response (resultCode: SUCCESS, success:true) to mobile App
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("user_modules/sequences/actions")
+local utils = require('user_modules/utils')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local function getPromptValue(pText)
+  return {
+    {
+      text = pText,
+      type = "TEXT"
+    }
+  }
+end
+
+local initialPromptValue = getPromptValue(" Make your choice ")
+
+local helpPromptValue = getPromptValue(" Help Prompt ")
+
+local timeoutPromptValue = getPromptValue(" Time out ")
+
+--[[ Local Functions ]]
+local function setChoiceSet(pChoiceIDValue)
+  local temp = {
+    {
+      choiceID = pChoiceIDValue,
+      menuName = "Choice" .. tostring(pChoiceIDValue),
+      vrCommands = {
+        "VrChoice" .. tostring(pChoiceIDValue),
+      }
+    }
+  }
+  return temp
+end
+
+local function createInteractionChoiceSet(pChoiceSetID)
+  local cid = common.getMobileSession():SendRPC("CreateInteractionChoiceSet", {
+      interactionChoiceSetID = pChoiceSetID,
+      choiceSet = setChoiceSet(pChoiceSetID),
+    })
+  common.getHMIConnection():ExpectRequest("VR.AddCommand", {
+      cmdID = pChoiceSetID,
+      type = "Choice",
+      vrCommands = { "VrChoice" .. tostring(pChoiceSetID) }
+    })
+  :Do(function(_, data)
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+    end)
+  common.getMobileSession():ExpectResponse(cid, { resultCode = "SUCCESS", success = true })
+end
+
+local function sendPI(pChoiceVR, pExpResult)
+  local params1 = {
+    initialText = "StartPerformInteraction",
+    initialPrompt = initialPromptValue,
+    interactionMode = "VR_ONLY",
+    interactionChoiceSetIDList = { 100 },
+    helpPrompt = helpPromptValue,
+    timeoutPrompt = timeoutPromptValue,
+    timeout = 5000,
+    vrHelp = { { text = pChoiceVR, position = 1 } }
+  }
+  local cid = common.getMobileSession():SendRPC("PerformInteraction", params1)
+  local success = true
+  if pExpResult ~= "SUCCESS" then success = false end
+  common.getMobileSession():ExpectResponse(cid, { success = success, resultCode = pExpResult })
+end
+
+local function PI_ViaVR_ONLY()
+  sendPI("ChoiceVR1", "SUCCESS")
+  common.run.runAfter(function() sendPI("ChoiceVR2", "REJECTED") end, 500)
```

Comment:
Seems this tests a different issue than https://github.com/smartdevicelink/sdl_core/issues/3838

Reply:
@jacobkeeler Yes, it's extended coverage for the issue https://github.com/smartdevicelink/sdl_hmi/pull/656 to check that SDL does not send UI.ClosePopUp to HMI in case mobile App sends two PerformInteraction requests with VR_ONLY interaction mode to SDL

Reply:
Hmmm, I'm tempted to say that it should be a separate issue, since it is not exclusive to `VR_ONLY` interactions (it should happen with BOTH as well).

Reply:
We can still keep both fixes in the same PR though